src/submission/submission-updates-messenger.ts

```diff
@@ -21,48 +21,63 @@ class DummyMessenger implements SubmissionUpdatesMessenger {
 }
 
 class KafkaMessenger implements SubmissionUpdatesMessenger {
-  private producer: Producer;
-  private knownTopics: ExpectedTopicsMap;
+  private kafka: Kafka;
+  private programUpdateTopic: TopicConfig;
 
-  constructor(clientId: string, brokers: string[], topics: ExpectedTopicsMap) {
-    this.producer = new Kafka({ clientId, brokers }).producer();
-    this.knownTopics = topics;
+  constructor(clientId: string, brokers: string[], topics: TopicConfig) {
+    this.kafka = new Kafka({ clientId, brokers });
+    this.programUpdateTopic = topics;
   }
 
   sendProgramUpdatedMessage = async (programId: string) => {
     L.info('KafkaMessenger called to send message to broker for updated program: ' + programId);
-    await this.producer.connect().catch(this.producerErrorHandler);
-    await this.producer
+
+    await this.createMissingProgramUpdateTopic().catch(this.errorHandler);
```

Comment:
shouldn't this be part of the bootstrap ?

Reply:
but, is it a one time thing though what if the topic doesn't exist when clinical tries to produce?

Reply:
it is a one time thing, like databases, why would we assume that topics would disappear? did you notice that while testing ?

Reply:
yea that makes sense will change :+1: